addons/website_sale/models/product.py

```diff
@@ -200,7 +200,7 @@ def _website_price(self):
         ret = self.env.user.has_group('sale.group_show_price_subtotal') and 'total_excluded' or 'total_included'
 
         for p, p2 in zip(self, self2):
-            taxes = partner.property_account_position_id.map_tax(p.taxes_id).filtered(lambda x: x.company_id == company_id)
+            taxes = partner.property_account_position_id.map_tax(p.sudo().taxes_id.filtered(lambda x: x.company_id == company_id))
```

Comment:
@pedrobaeza Yes, the problem still shows up with one company.

Steps:

1. Enable b2c prices
2. Add customer tax to product.
3. With public user, error 500 (i put 403 before wich wasn't right, sorry) log:
```
AccessError: (u'Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:\n- Employees/Employee\n\t- Accounting & Finance/Billing\n\t- Accounting & Finance/Adviser\n\t- Sales/User: Own Documents Only\n\t- Sales/User: Own Documents Only\n\t- Other Extra Rights/Portal\n\n(Document model: account.tax)', None)
```
4. No issues being other type of user.

So I'll research now in another PR the minimum ACL needed for public user to give public user such permissions and leave this one pending until that one is ackwoledge.